python/whylogs/experimental/extras/confighub.py

```diff
@@ -0,0 +1,130 @@
+import logging
+import os
+from abc import ABC, abstractmethod
+from typing import List, Optional
+
+from semantic_version import Version
+
+logger = logging.getLogger(__name__)
+
+
+class ConfigStore(ABC):
+    def __init__(self, org_id: str, dataset_id: str, what: str):
+        """
+        what in {"constraints", "validators", ... }
+        """
+        self._org_id = org_id
+        self._dataset_id = dataset_id
+        self._what = what
+
+    @abstractmethod
+    def get_latest(self) -> str:
+        """Returns latest version of the config"""
+        pass
+
+    def get_version_of_latest(self) -> Version:
+        """
+        Returns the Version of the current config
+        """
+        return self.get_available_versions()[-1]
+
+    @abstractmethod
+    def get_version(self, ver: Version) -> str:
+        """
+        Returns the requested version of the config. ValueError on invalid/unavailable.
+        """
+        pass
+
+    @abstractmethod
+    def get_available_versions(self) -> List[Version]:
+        """
+        Returns available versions in ascending order
+        """
+        pass
+
+    @abstractmethod
+    def propose_version(self, config: str, new_version: Version, note: Optional[str] = None) -> None:
+        """
```

Comment:
I think as currently written this can only be called once for a new version. Should I allow multiple calls to update the new version until it's committed (or dropped)? Should there be a separate `update_version(self, config, new_version, note)` method for updating an existing new version until it's committed (or dropped)?

Reply:
Actually this might work for multiple calls.